src/FSharpPlus/Control/Applicative.fs

```diff
@@ -143,21 +143,92 @@ type Lift3 with
     static member inline Lift3 (_, (_:'t when 't: null and 't: struct, _: ^u when ^u : null and ^u: struct, _: ^v when ^v : null and ^v: struct), _mthd: Default1) = id
     static member inline Lift3 (f: 'T -> 'U -> 'V -> 'W, (x: '``Applicative<'T>``, y: '``Applicative<'U>``, z: '``Applicative<'V>``)            , _mthd: Default1) = ((^``Applicative<'T>`` or ^``Applicative<'U>`` or ^``Applicative<'V>`` ) : (static member Lift3 : _*_*_*_ -> _) f, x, y, z)
 
+type IsLeftZeroHelper =
+    static member Seq (xs: seq<'t>) = Seq.isEmpty xs
+    static member NonEmptySeq (_: NonEmptySeq<'t>) = false
+    static member List (xs: list<'t>) = List.isEmpty xs
+    static member Array (xs: array<'t>) = Array.isEmpty xs
+    static member Option (x: option<'t>) = Option.isNone x
+    static member Result (x: Result<'t, 'u>) = match x with Error _ -> true | _ -> false
+    static member Choice (x: Choice<'t, 'u>) = match x with Choice2Of2 _ -> true | _ -> false
+
+#if !FABLE_COMPILER
+type IsLeftZeroHelper<'a>() =
+    /// turns false if
+    /// - it should always return false because neither `IsLeftZero` nor `Empty` are present
+    /// - the target method is inlined and cannot be called through reflection
+    static let mutable isValid = true
```

Comment:
That's an interesting trick 😃

Reply:
yeah, it seems the compiler does not add any kind of metadata to the uncallable inline methods 😅